lib/logstash/codecs/auto_flush.rb

```diff
@@ -11,16 +11,21 @@ def start
     # can't start if pipeline is stopping
     return self if stopped?
     if pending?
-      @task.reset
+      @task.cancel
```

Comment:
> Firstly, I just realised a bug: I don't cancel the task when we flush normally this could flush a half built buffer if the lines come along slowly.

Correct good catch!